R/zzz.R

```diff
@@ -22,17 +22,7 @@ register_mlr3 = function(libname, pkgname) {
 }
 
 .onLoad = function(libname, pkgname) { # nolint
-  register_mlr3()
-  setHook(packageEvent("mlr3", "onLoad"), function(...) register_mlr3(),
-    action = "append")
-}
-
-.onUnload = function(libpath) { # nolint
-  event = packageEvent("mlr3", "onLoad")
-  hooks = getHook(event)
-  pkgname = vapply(hooks, function(x) environment(x)$pkgname, NA_character_)
-  setHook(event, hooks[pkgname != "mlr3extralearners"],
-    action = "replace")
+  mlr3misc::register_namespace_callback(pkgname, "mlr3", register_mlr3)
```

Comment:
Just to confirm, this function is only for .onLoad and nothing for .onUnload?

Reply:
yes, it automatically adds hooks for the `onUnload` event.

Reply:
Thanks